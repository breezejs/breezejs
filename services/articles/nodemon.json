{
  "ext": "ts",
  "ignore": [
    ".e2e-spec.ts",
    ".spec.ts"
  ],
  "watch": [
    "src"
  ]
}
