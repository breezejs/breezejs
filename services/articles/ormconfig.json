[
  {
    "entities": [
      "node_modules/@breezejs/sql/lib/entities/article/article.entity.js",
      "node_modules/@breezejs/sql/lib/entities/media/**/*.entity.js",
      "node_modules/@breezejs/sql/lib/entities/user/user.entity.js"
    ],
    "database": "development",
    "host": "localhost",
    "logging": true,
    "password": "password",
    "port": 3306,
    "synchronize": true,
    "type": "mysql",
    "username": "breeze"
  },
  {
    "entities": [
      "node_modules/@breezejs/sql/lib/entities/article/article.entity.js",
      "node_modules/@breezejs/sql/lib/entities/media/**/*.entity.js",
      "node_modules/@breezejs/sql/lib/entities/user/user.entity.js"
    ],
    "database": "testing",
    "host": "localhost",
    "logging": false,
    "name": "test",
    "password": "password",
    "port": 3306,
    "synchronize": true,
    "type": "mysql",
    "username": "breeze"
  }
]
