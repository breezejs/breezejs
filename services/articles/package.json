{
  "name": "@breezejs/articles",
  "version": "0.0.0",
  "description": "Breeze articles API",
  "author": "Mo Gusbi <me@mogusbi.co.uk>",
  "homepage": "https://github.com/breezejs/breezejs",
  "license": "MIT",
  "main": "lib/main.js",
  "typings": "lib/main.d.ts",
  "directories": {
    "lib": "lib"
  },
  "files": [
    "lib"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/breezejs/breezejs.git"
  },
  "scripts": {
    "build": "tsc",
    "docs": "typedoc --options ../../typedoc.js --out ./docs ./src",
    "e2e": "jest --config=e2e/jest.config.js --runInBand --forceExit",
    "lint": "tslint -c ../../tslint.json -p tsconfig.lint.json",
    "start": "nodemon --exec ts-node src/main.ts",
    "test": "jest --coverage",
    "posttest": "codecov"
  },
  "dependencies": {
    "@breezejs/article": "^0.0.0",
    "@breezejs/bootstrap": "^0.0.0",
    "@breezejs/request": "^0.0.0",
    "mysql": "^2.16.0",
    "typeorm": "^0.2.14"
  },
  "jest": {
    "collectCoverageFrom": [
      "**/src/**/*.ts",
      "!**/e2e/**/*.ts",
      "!**/src/**/*.dto.ts",
      "!**/src/**/*.entity.ts",
      "!**/src/**/*.model.ts",
      "!**/src/**/*.module.ts",
      "!**/src/**/*.providers.ts"
    ],
    "coveragePathIgnorePatterns": [
      "index.ts",
      "main.ts",
      "/node_modules/"
    ],
    "moduleFileExtensions": [
      "ts",
      "js",
      "json",
      "node"
    ],
    "moduleNameMapper": {
      "@breezejs/article": "<rootDir>/node_modules/@breezejs/article/lib/__mocks__"
    },
    "testMatch": [
      "**/+(*.)+(spec|test).+(ts|js)?(x)"
    ],
    "transform": {
      "^.+\\.tsx?$": "ts-jest"
    }
  }
}
