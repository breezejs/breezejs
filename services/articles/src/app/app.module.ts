/**
 * @author Mo Gusbi <me@mogusbi.co.uk>
 */
import {Module} from '@nestjs/common';
import {ArticlesModule} from './articles';

@Module({
  imports: [
    ArticlesModule
  ]
})
export class AppModule {}
