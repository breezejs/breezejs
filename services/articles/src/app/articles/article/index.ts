/**
 * @author Mo Gusbi <me@mogusbi.co.uk>
 */
export * from './article.controller';
