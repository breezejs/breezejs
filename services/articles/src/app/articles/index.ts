/**
 * @author Mo Gusbi <me@mogusbi.co.uk>
 */
export * from './articles.module';
