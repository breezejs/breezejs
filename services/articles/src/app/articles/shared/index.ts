/**
 * @author Mo Gusbi <me@mogusbi.co.uk>
 */
export * from './article.dto';
