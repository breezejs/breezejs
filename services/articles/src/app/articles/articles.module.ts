/**
 * @author Mo Gusbi <me@mogusbi.co.uk>
 */
import {SqlModule} from '@breezejs/sql';
import {Module} from '@nestjs/common';
import {ArticleController} from './article';
import {ArticleProviders, ArticleService} from './shared';

/**
 * Articles module
 */
@Module({
  controllers: [
    ArticleController
  ],
  imports: [
    SqlModule
  ],
  providers: [
    ...ArticleProviders,
    ArticleService
  ]
})
export class ArticlesModule {}
