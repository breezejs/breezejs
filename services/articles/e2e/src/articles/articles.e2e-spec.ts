/**
 * @author Mo Gusbi <me@mogusbi.co.uk>
 */
import {INestApplication, ValidationPipe} from '@nestjs/common';
import {Test, TestingModule} from '@nestjs/testing';
import * as supertest from 'supertest';
import {ArticlesModule} from '../../../src/app/articles';

describe('Articles', (): void => {
  let app: INestApplication;
  // let id: string = null;
  let request: supertest.SuperTest<supertest.Request>;

  beforeAll(async (): Promise<void> => {
    const testModule: TestingModule = await Test
      .createTestingModule({
        imports: [
          ArticlesModule
        ]
      })
      .compile();

    app = testModule.createNestApplication();

    app.useGlobalPipes(new ValidationPipe());

    await app.init();

    request = supertest(app.getHttpServer());
  });

  afterAll(async (): Promise<void> => {
    await app.close();
  });

  // describe('[GET] /articles', (): void => {});

  describe('[GET] /articles/:id', (): void => {
    it('should return a article', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.get('/articles/38015b30-9194-4957-a3eb-00829c4d52c7');

      expect(status).toEqual(200);
      expect(body).toMatchSnapshot();
    });

    it('should return a filtered result', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.get('/articles/38015b30-9194-4957-a3eb-00829c4d52c7').query({
        fields: 'article.title,article.content'
      });

      expect(status).toEqual(200);
      expect(body).toMatchSnapshot();
    });

    it('should return a bad request if filtered field does not exist', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.get('/articles/38015b30-9194-4957-a3eb-00829c4d52c7').query({
        fields: 'article.name'
      });

      expect(status).toEqual(400);
      expect(body).toMatchSnapshot();
    });

    it('should return not found if competition does not exist', async (): Promise<void> => {
      const {body, status}: supertest.Response = await request.get('/articles/70340225-c90a-4b9b-a979-2d16212acd16');

      expect(status).toEqual(404);
      expect(body).toMatchSnapshot();
    });
  });

  // describe('[POST] /articles', (): void => {});
  //
  // describe('[PATCH] /articles/:id', (): void => {});

  // describe('[DELETE] /articles/:id', (): void => {
  //   it('should delete a competition', async (): Promise<void> => {
  //     const {body, status}: supertest.Response = await request.delete(`/articles/${id}`);
  //
  //     expect(status).toEqual(204);
  //     expect(body).toEqual({});
  //   });
  //
  //   it('should return not found if competition does not exist', async (): Promise<void> => {
  //     const {body, status}: supertest.Response = await request.delete('/articles/88044266-a83a-4e10-b85b-e552790909a1');
  //
  //     expect(status).toEqual(404);
  //     expect(body).toEqual({
  //       error: 'Not Found',
  //       statusCode: 404
  //     });
  //   });
  // });
});
