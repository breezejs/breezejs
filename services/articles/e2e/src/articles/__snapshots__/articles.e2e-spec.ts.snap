// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles [GET] /articles/:id should return a article 1`] = `
Object {
  "alternativeTitle": null,
  "author": Object {
    "createdAt": "2019-02-10T00:00:00.000Z",
    "emailAddress": "me@mogusbi.co.uk",
    "forename": "Mo",
    "id": "f07a1ccb-f3bf-4ac8-beab-c740668674c4",
    "surname": "Gusbi",
    "updatedAt": "2019-02-10T00:00:00.000Z",
  },
  "content": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla fringilla venenatis urna ut fringilla.",
  "createdAt": "2019-02-08T00:00:00.000Z",
  "id": "38015b30-9194-4957-a3eb-00829c4d52c7",
  "media": Object {
    "copyright": null,
    "createdAt": "2019-02-09T00:00:00.000Z",
    "description": "Image 11",
    "id": "f5ed1aa8-313c-4afd-9978-f47d81b7d23c",
    "source": Array [
      Object {
        "id": "00117b67-e296-47ac-8676-27a519de373d",
        "name": "thumbnail",
        "path": "https://via.placeholder.com/480x380/555/FFF?text=Placeholder",
      },
      Object {
        "id": "dac73a4e-6fb6-4daa-8cb2-0b0b1b9ecb10",
        "name": "main",
        "path": "https://via.placeholder.com/1200x630/555/FFF?text=Placeholder",
      },
    ],
    "updatedAt": "2019-02-09T00:00:00.000Z",
  },
  "publishDate": "2019-02-08T00:00:00.000Z",
  "slug": "2019/2/8/second-article-that-has-been-created",
  "teaser": null,
  "title": "Second article that has been created",
  "updatedAt": "2019-02-08T00:00:00.000Z",
}
`;

exports[`Articles [GET] /articles/:id should return a bad request if filtered field does not exist 1`] = `
Object {
  "error": "Bad Request",
  "message": "ER_BAD_FIELD_ERROR: Unknown column 'article.name' in 'field list'",
  "statusCode": 400,
}
`;

exports[`Articles [GET] /articles/:id should return a filtered result 1`] = `
Object {
  "content": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla fringilla venenatis urna ut fringilla.",
  "title": "Second article that has been created",
}
`;

exports[`Articles [GET] /articles/:id should return not found if competition does not exist 1`] = `
Object {
  "error": "Not Found",
  "statusCode": 404,
}
`;
